### ISSUE TITLE
"save before continuing" after clicking "done" on a page

### ISSUE THREAD

**Issue by saracarl** [benwbrum/fromthepage#3715]
A transcriber for NEHGS experienced the following:

> Hi - One of our volunteers has a question/problem that I am not sure how to answer. This is what they said in their email - "I completed a couple of small pages. I hit "Save" and "Done" then tried to go to the next page. I got a message that said to save before continuing and it wouldn't let me do anything. I kept hitting Save and nothing happened. I couldn't even toggle in and out of full screen mode. I had to quit and start over."

Here's their ahoy log:
https://fromthepage.com/admin/visit_actions?user_id=32016466&visit_id=32667774

Focusing on the last "save" (assuming that's the one they got frustrated on and gave up) I pulled logs.  The session (+ other stuff in the same thread) is on prod here:
fromthepage@fromthepage:~/deployment/current/log$ ls -l 19689.log 
-rw-rw-r-- 1 fromthepage fromthepage 14691947 Jul 20 15:24 19689.log

Here's what I think is the logging around that save:

```
I, [2023-07-19T17:48:12.914023 #19689]  INFO -- : Processing by TranscribeController#save_transcription as HTML
I, [2023-07-19T17:48:12.914312 #19689]  INFO -- :   Parameters: {"authenticity_token"=>"zB4o1S8wxYXr09i2Gjrw42vrbST37teitZdWE4DxnxlroWTM20PgQbMGU
9bbr1VKCHi/kNlZaC/+ecg0XjJpOw==", "page_id"=>"33141837", "flow"=>"", "quality_sampling_id"=>"", "auto-fullscreen"=>"yes", "page"=>{"mark_blank"=>
"0"}, "save_to_needs_review"=>"", "filter-brightness"=>"0", "filter-contrast"=>"0", "filter-threshold"=>"0", "fields"=>{"5339"=>{"Indexing Spread
sheet"=>"[[\"\",\"\",\"\",\"BELDEN\",\"Frederick H.\",\"5y8m\",\"Residence\",\"\",\"\",\"\",\"\",\"Portsmouth\",\"Rockingham\",\"New Hampshire\",
\"United States\",\"\",\"\"],[\"\",\"\",\"\",\"DODGE\",\"Leona M.\",\"15y8m\",\"Residence\",\"\",\"\",\"\",\"\",\"Portsmouth\",\"Rockingham\",\"N
ew Hampshire\",\"United States\",\"\",\"\"],[\"\",\"\",\"\",\"HISLOP\",\"Gordon D.\",\"5y7m\",\"Residence\",\"\",\"\",\"\",\"\",\"Portsmouth\",\"
Rockingham\",\"New Hampshire\",\"United States\",\"\",\"\"],[\"\",\"\",\"\",\"HODGKINS\",\"Ruth M.\",\"15y2m\",\"Residence\",\"\",\"\",\"\",\"\",
\"Portsmouth\",\"Rockingham\",\"New Hampshire\",\"United States\",\"\",\"\"],[\"\",\"\",\"\",\"PEYSER\",\"Virginia\",\"4y3m\",\"Residence\",\"\",
\"\",\"\",\"\",\"Portsmouth\",\"Rockingham\",\"New Hampshire\",\"United States\",\"\",\"\"],[\"\",\"\",\"\",\"TAYLOR\",\"Constance J.\",\"6y3m\",
\"Residence\",\"\",\"\",\"\",\"\",\"Portsmouth\",\"Rockingham\",\"New Hampshire\",\"United States\",\"\",\"\"],[null,null,null,null,null,null,nul
l,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null
,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,nu
ll,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,nul
l,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null
,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,nu
ll,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,nul
l,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null
,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,nu
ll,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,nul
l,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null
,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,nu
ll,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,nul
l,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null
,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,nu
ll,null,null,null,null]]"}, "5340"=>{"Portsmouth, Rockingham, New Hampshire, United States"=>""}}, "user_slug"=>"nehgsadmin007", "collection_id"=
>"portsmouth-nh-school-records"}
I, [2023-07-19T17:48:12.944473 #19689]  INFO -- : TRANSCRIPTION 2023-07-19 17:48:12 +0000
I, [2023-07-19T17:48:12.990219 #19689]  INFO -- : TRANSCRIPTION 2023-07-19 17:48:12 +0000       SUCCESS 
I, [2023-07-19T17:48:13.244492 #19689]  INFO -- : Redirected to https://fromthepage.com/transcribe/assign_categories?collection_id=portsmouth-nh-
school-records&page_id=33141837
I, [2023-07-19T17:48:13.250858 #19689]  INFO -- : Completed 302 Found in 336ms (ActiveRecord: 222.8ms | Allocations: 116131)
I, [2023-07-19T17:48:13.251171 #19689]  INFO -- : Oink Action: transcribe#save_transcription
I, [2023-07-19T17:48:13.251340 #19689]  INFO -- : Memory usage: 9765084 | PID: 19689
I, [2023-07-19T17:48:13.251484 #19689]  INFO -- : Instantiation Breakdown: Total: 1023 | Work: 444 | WorkStatistic: 443 | TableCell: 103 | Spread
sheetColumn: 17 | Page: 3 | Collection: 3 | User: 2 | TranscriptionField: 2 | PageVersion: 2 | PageBlock: 1 | FriendlyId::Slug: 1 | Visit: 1 | Ah
oy::Event: 1
I, [2023-07-19T17:48:13.251606 #19689]  INFO -- : Oink Log Entry Complete
I, [2023-07-19T17:48:13.312090 #19689]  INFO -- : Started GET "/transcribe/assign_categories?collection_id=portsmouth-nh-school-records&page_id=3
3141837" for 98.216.212.31 at 2023-07-19 17:48:13 +0000
I, [2023-07-19T17:48:13.314081 #19689]  INFO -- : Processing by TranscribeController#assign_categories as HTML
I, [2023-07-19T17:48:13.314335 #19689]  INFO -- :   Parameters: {"collection_id"=>"portsmouth-nh-school-records", "page_id"=>"33141837"}
I, [2023-07-19T17:48:13.325573 #19689]  INFO -- :   Rendering inline template
I, [2023-07-19T17:48:13.326940 #19689]  INFO -- :   Rendered inline template (Duration: 1.0ms | Allocations: 83)
I, [2023-07-19T17:48:13.327568 #19689]  INFO -- :   Rendering inline template
I, [2023-07-19T17:48:13.328409 #19689]  INFO -- :   Rendered inline template (Duration: 0.5ms | Allocations: 79)
I, [2023-07-19T17:48:13.328909 #19689]  INFO -- :   Rendering inline template
I, [2023-07-19T17:48:13.329590 #19689]  INFO -- :   Rendered inline template (Duration: 0.4ms | Allocations: 79)
I, [2023-07-19T17:48:13.334017 #19689]  INFO -- : Redirected to https://fromthepage.com/nehgsadmin007/portsmouth-nh-school-records/sdr-unknown-1915-1916/transcribe/33141837
I, [2023-07-19T17:48:13.341719 #19689]  INFO -- : Completed 302 Found in 27ms (ActiveRecord: 5.4ms | Allocations: 13676)
I, [2023-07-19T17:48:13.342039 #19689]  INFO -- : Oink Action: transcribe#assign_categories
I, [2023-07-19T17:48:13.342223 #19689]  INFO -- : Memory usage: 9765084 | PID: 19689
I, [2023-07-19T17:48:13.342394 #19689]  INFO -- : Instantiation Breakdown: Total: 14 | PageBlock: 6 | User: 2 | Collection: 2 | Page: 1 | Work: 1 | Visit: 1 | Ahoy::Event: 1
I, [2023-07-19T17:48:13.342534 #19689]  INFO -- : Oink Log Entry Complete
I, [2023-07-19T17:48:13.501341 #19689]  INFO -- : Started GET "/article/show?article_id=54388" for 185.191.171.16 at 2023-07-19 17:48:13 +0000
I, [2023-07-19T17:48:13.503127 #19689]  INFO -- : Processing by ArticleController#show as HTML
I, [2023-07-19T17:48:13.503329 #19689]  INFO -- :   Parameters: {"article_id"=>"54388"}
I, [2023-07-19T17:48:13.504450 #19689]  INFO -- : [ahoy] Visit excluded
I, [2023-07-19T17:48:13.520359 #19689]  INFO -- :   Rendered article/_graph.dot.erb (Duration: 1.5ms | Allocations: 1565)
I, [2023-07-19T17:48:13.548918 #19689]  INFO -- :   Rendering article/show.html.slim within layouts/application
I, [2023-07-19T17:48:13.551253 #19689]  INFO -- :   Rendered shared/_article_tabs.html.slim (Duration: 1.9ms | Allocations: 1131)
I, [2023-07-19T17:48:13.569878 #19689]  INFO -- :   Rendered article/_article_links.html.slim (Duration: 13.8ms | Allocations: 5790)
I, [2023-07-19T17:48:13.570873 #19689]  INFO -- :   Rendered article/show.html.slim within layouts/application (Duration: 21.5ms | Allocations: 9787)
I, [2023-07-19T17:48:13.572429 #19689]  INFO -- :   Rendered layouts/_mixpanel.erb (Duration: 0.1ms | Allocations: 34)
I, [2023-07-19T17:48:13.573203 #19689]  INFO -- :   Rendered layouts/_fb_pixel.erb (Duration: 0.1ms | Allocations: 34)
I, [2023-07-19T17:48:13.576226 #19689]  INFO -- :   Rendered shared/_breadcrumbs.html.slim (Duration: 0.2ms | Allocations: 87)
I, [2023-07-19T17:48:13.577236 #19689]  INFO -- :   Rendered shared/_user_collection_footer.html.slim (Duration: 0.1ms | Allocations: 39)
I, [2023-07-19T17:48:13.579195 #19689]  INFO -- :   Rendered layouts/_ga.erb (Duration: 0.1ms | Allocations: 33)
I, [2023-07-19T17:48:13.596229 #19689]  INFO -- : [ahoy] Event excluded
I, [2023-07-19T17:48:13.597203 #19689]  INFO -- : Completed 200 OK in 94ms (Views: 40.5ms | ActiveRecord: 15.1ms | Allocations: 32030)
```

My observations:

- The ahoy save timestamp is 2 seconds after this save -- not sure if that's relevant, but this was the closest.
- Why are we routing this save through a subject linking action?  Do they have subject linking turned on?  Do they intend to?  (Maybe, because we turn it off by default.  That or we're doing extraneous work on the save.)
- I can't see from the logs how to differentiate a "save" from a "done", which I think would be key to differentiating the log messages for what the transcriber reported.

**Comment by saracarl** [comment]
Here's the collection:
https://fromthepage.com/nehgsadmin007/portsmouth-nh-school-records

Confirmed that subject linking is not enabled.

**Comment by benwbrum** [comment]
On the questions, 
* All saves route through `assign_categories`, which automatically redirects if there are no subjects
* I see the parameter `save_to_needs_review` so it looks like this is a spreadsheet project set up with review required.

**Comment by saracarl** [comment]
I transcribed a page in this project, saved, then "doned".  No problems.  Clicked "next" arrow, no problem.  Typed a space in the notes field, get a *different* message when I try to "next", so it's not that.  When I type something into the spreadsheet and try to "next" before saving, then I get the "save before continuing" message.  Reasonable, but implies that we "think" there's new data in the spreadsheet that isn't saved when the transcriber clicks "next".  

There's no `goto_next_untranscribed_page` call for this work/collection in the logs I pulled, so I think we're checking before we call that and bailing out because of unsaved data.

Could it be a network delay?  Too long to save before she clicks next?
Could it be her session expiring or something similar?  I don't think so.

Looked at her page versions, not very useful:
https://fromthepage.com/page_version/show?page_version_id=35385508
Other than there's a lot of page saves here.
